src/lib.rs

```diff
@@ -1142,4 +1209,86 @@ mod tests {
         assert_eq!(di.size_hint(), (2, Some(2)));
         assert_eq!(di.count(), 2);
     }
+
+    #[test]
+    fn test_subtract_debug_and_size_hint() {
+        let vikings: HashBag<&'static str> = ["Einar", "Olaf", "Harald"].iter().cloned().collect();
+        let killed_vikings: HashBag<&'static str> = ["Einar"].iter().cloned().collect();
+        let mut alive_vikings = vikings.subtract(&killed_vikings);
+        println!("{:?}", alive_vikings);
+
+        assert_eq!(alive_vikings.size_hint(), (0, Some(3)));
+
+        // Note that we can't assume in what order the vikings will come, only
+        // that there shall be Some(_) viking two times
+        alive_vikings.next().unwrap();
```

Comment:
Would you mind also adding a `size_hint` check in the middle here so that we can test that the upper bound indeed goes down for each iteration?

Reply:
I don't think we can do that (as indicated by my comment) because we don't know in what order elements are iterated. Although... if I add a new test with two disjoint sets I think the size hint becomes predictable. I will do that.

Reply:
Done.